frontend/app/controllers/Joiner.scala

```diff
@@ -132,6 +132,10 @@ object Joiner extends Controller with ActivityTracking
     (for {
       identityUser <- identityService.getIdentityUserView(request.user, identityRequest)
     } yield {
+      tier match {
+        case t: Tier.Supporter => MembersDataAPI.Service.addBehaviour(request, "enterPaidDetails.show")
+        case _ =>
```

Comment:
It's an acceptable thing to do: http://stackoverflow.com/questions/25843677/what-does-it-mean-to-have-an-empty-case-statement-in-scala Admittedly we're not returning anything here, but we're saying explicitly that we've considered what to do with a tier that isn't of the type we're looking for.